sphinx/ext/autodoc/__init__.py

```diff
@@ -2152,7 +2152,8 @@ def add_content(self, more_content: StringList | None) -> None:
         if not more_content:
             more_content = StringList()
 
-        self.update_content(more_content)
+        if not self.get_module_comment(self.objpath[-1]):
```

Comment:
The issue with this path is that it entirely disallows a documenter to subclass `update_content`. An alternative is to provide a method, say `should_update_content` which returns a boolean determining whether the directive should update the content or not. By default, this would return `not self.get_module_comment(self.objpath[-1])` and subclasses could decide what to do.

Reply:
This is handled the same way in `ClassDocumenter`

https://github.com/sphinx-doc/sphinx/blob/45f3cd6324e4d6efea2dec03721b2946c0cbd32c/sphinx/ext/autodoc/__init__.py#L1861

So shall we keep it this way for now?